File: backend/notification-service/src/main/java/com/ecobridge/notification/service/NotificationService.java
package com.ecobridge.notification.service;

import com.ecobridge.notification.entity.Notification;

import java.util.List;
import java.util.UUID;

public interface NotificationService {

    void createNotification(
            UUID userId,
            String title,
            String message,
            String type
    );

    List<Notification> getNotifications(UUID userId);

    long getUnreadCount(UUID userId);

    void markAsRead(UUID id);

    void markAllAsRead(UUID userId);

    void deleteNotification(UUID id);

}

File: frontend/ecobridge-frontend/src/services/notificationService.js
import api from "../api/notificationApi";

export const getNotifications = (userId) => {
  return api.get(`/api/v1/notifications/${userId}`);
};

export const getUnreadCount = (userId) => {
  return api.get(`/api/v1/notifications/${userId}/unread-count`);
};

export const markAsRead = (id) => {
  return api.put(`/api/v1/notifications/${id}/read`);
};

export const markAllAsRead = (userId) => {
  return api.put(`/api/v1/notifications/${userId}/read-all`);
};

export const deleteNotification = (id) => {
  return api.delete(`/api/v1/notifications/${id}`);
};